File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
import java.util.Arrays;

class Solution {
    public int solution(int[] arr) {
        int count = 0;
        int[] temp = {};
        while(true){
            temp = Arrays.copyOf(arr, arr.length);
            for(int i = 0; i < arr.length; i++){
                if(arr[i] >= 50 && arr[i] % 2 == 0){
                    arr[i] /= 2;
                }
                else if(arr[i] < 50 && arr[i] % 2 != 0){
                    arr[i] = arr[i] * 2 + 1;
                }
            }
            if(Arrays.equals(arr, temp)) return count;
            
            count++;
        }
    }
}

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.js
function solution(arr) {
    let count = 0;
    let temp = {};
    while(true){
        temp = arr.slice();
        for(let i = 0; i < arr.length; i++){
            if(arr[i] >= 50 && arr[i] % 2 == 0){
                arr[i] /= 2;
            }
            else if(arr[i] < 50 && arr[i] % 2 != 0){
                arr[i] = arr[i] * 2 + 1;
            }
        }
        if(JSON.stringify(temp) === JSON.stringify(arr)) return count;
        count++;
    }
}